### ISSUE TITLE
Moving our Community Fourms Server to a Team Account on Digital Ocean

### ISSUE THREAD

**Issue by odysseywestra** [mypaint/mypaint.github.io#32]
As some may know, our community forums are hosted on a [Digital Ocean Droplet](https://www.digitalocean.com/) and are sponsored by [Altispeed](http://www.altispeed.com/). I proposed to them to move our Community Fourms to a [Team Account](https://www.digitalocean.com/community/tutorials/how-to-use-teams-on-digitalocean). This would allow us to expand if need be, and if need be enable us to easily switch funding if the time comes. 

Noah, the head of Altispeed asked me to write up a plan of what we would like to do and what kind of budget we would need. One of the thing I proposed was spinning up a [Mattermost ](https://about.mattermost.com/)server so we can have a area to provide real time support and allow contributors to communicate in realtime. Another thing would be to create build servers that can be spin up on demand and allow contributors to provide test builds to make things like for pull requests much easier to test. Plus it is possible to run [windows servers](https://www.whatuptime.com/installing-microsoft-windows-onto-digitalocean-droplet/) on digital ocean which should make building on windows easier if need be too. Plus we can have a more permanent spot for our Website which is currently hosted on [Github Pages](https://pages.github.com/).

These servers can be spin up and destroyed on demand so they don't have to be on 24/7 and won't cost a lot of money.

So any ideas of how we could take advantage of a Digital Ocean Team Account for the MyPaint Community team? Plus with those ideas what kind of budget would we need to accomplish them?

**Comment by odysseywestra** [comment]
Another thing we can do is also run our own instance of transfer.sh and have a bot post a link to the builds of a Pull Request that Travis-CI makes  so users can test them out easier.

**Comment by achadwick** [comment]
What is the driver behind this? I have issues.

I have no idea how this would be funded. How are you proposing to finance this extremely wide-reaching change? I also see no benefit to making these changes to the project infrastructure right now.

We probably don't need a Windows build server: our Windows builds can be stitched together from prebuilt binary packages for everything, including libmypaint. This means that they can be carried out by TeaCI (drone+Wine+MSYS2) or AppVeyor (spun-up Windows VMs with MSYS2 (and a lot of stuff that messes up MSYS2 SCons, which is why libmypaint uses it but mypaint doesn't right now)). I think that AppVeyor can be plumbed into GH releases or random upload sites.

We do have a Slack account, and could use that. Plus, everyone already has slack accounts. However I haven't had time or any indication that people are willing to contribute _their_ time to do support and project planning on it, however, so I haven't yet taken it forward. Slack or a slack-like system would be a good alternative to the old mailing lists or IRC, but very few people actually use those right now.

**Comment by odysseywestra** [comment]
With the build servers, I understand. I didn't look into the Continuous Integration tools till after I wrote this. Ad yes,we could use GH release, dropbox, drive, or transfer.sh as it now stands. 

What prompted me raise this issue is because as it now stands, Altispeed are the ones who spun and gave me ssh access to the server on Digital Ocean. If they were to not want to sponsor the server anymore, it would mean we would have to move the Community site to a new Digital Ocean account, or new server which would mean downtime. If we were to move the Community site to a team account, the site wouldn't have to move afterwards, and would open the possibility expanding as long as we stay within the budget of the sponsor. Plus it will make it easier for me to maintain and back it up and make tests before deploying them to the server. If by chance Altispeed were to drop support, it would make it easier for me to take up the costs without downtime. Plus if I were to step down, I can give someone else access easily.

On the subject of slack, I'm one of the people who doesn't use it so I don't have an account. I mainly communicate via [Telegram](https://telegram.org/) and email. There is a plugin however for Discourse called [Babble](https://meta.discourse.org/t/babble-a-chat-plugin/31753/565) that I've activated under the [General Chit Chat](https://community.mypaint.org/c/general). You just need to click on the Chat link on the Category Bar. 

I can enable that chat system for any of the categories like the user support and developers area. Since most of our users are on the Community forums that would make it easier for people to use it.
I'm also looking into to see if a two-way syncing of the Github Issue tracker and a category on our community forums.

So basically my main reason for moving the community forums to a Team Account is to make it easier to maintain it and also give us the opportunity to expand or test stuff if we need to.
